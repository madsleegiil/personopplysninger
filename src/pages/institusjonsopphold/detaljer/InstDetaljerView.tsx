import React from "react";
import ListElement from "components/listelement/ListElement";
import { FormattedMessage } from "react-intl";
import moment from "moment";
import { formatOrgnr, RADIX_DECIMAL } from "../../../utils/formattering";
import { InstInnslag } from "../../../types/inst";
import { BodyShort, Heading, HelpText } from "@navikt/ds-react";

const InstDetaljerView = (props: { innslag: InstInnslag }) => {
  const { innslag } = props;

  return (
    <div>
      <div className="detaljer__tittel">
        <Heading level="2" size="small">
          {innslag.institusjonsnavn}
        </Heading>
        {innslag.organisasjonsnummer && (
          <BodyShort>
            <FormattedMessage
              id="side.organisasjonsnummer"
              values={{
                orgnr: formatOrgnr(
                  parseInt(
                    innslag.organisasjonsnummer,
                    RADIX_DECIMAL
                  ).toString()
                ),
              }}
            />
          </BodyShort>
        )}
      </div>
      <hr className="box__linje-bred" />
      <div className="box">
        <div className="box__content">
          <dl className="list-column-2">
            <ListElement
              titleId={"inst.institusjonstype"}
              content={innslag.institusjonstype}
            />
            <ListElement
              titleId={"inst.periode"}
              content={
                <div className={"inst__periode"}>
                  {`${moment(innslag.startdato).format("DD.MM.YYYY")} - ${
                    innslag.faktiskSluttdato
                      ? moment(innslag.faktiskSluttdato).format("DD.MM.YYYY")
                      : ``
                  }`}
                  {innslag.fiktivSluttdato && (
                    <HelpText>
                      <FormattedMessage id={"inst.fiktivSluttdato"} />
                    </HelpText>
                  )}
                </div>
              }
            />
          </dl>
        </div>
      </div>
    </div>
  );
};

export default InstDetaljerView;
