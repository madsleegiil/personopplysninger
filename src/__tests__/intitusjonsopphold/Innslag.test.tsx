import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import instInfo from "../../clients/apiMock/app/fetch/inst-info.json";
import InstDetaljerView from "../../pages/institusjonsopphold/detaljer/InstDetaljerView";

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
  useLocation: () => ({
    pathname: "pathname",
  }),
}));

jest.mock("nav-frontend-hjelpetekst");

describe("InstDetaljerView", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale={"nb"} messages={nbMessages}>
          <InstDetaljerView innslag={instInfo[0]} />
        </IntlProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
