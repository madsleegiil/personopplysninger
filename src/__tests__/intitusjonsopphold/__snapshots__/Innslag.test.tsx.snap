// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstDetaljerView renders correctly 1`] = `
<div>
  <div
    className="detaljer__tittel"
  >
    <h2
      className="typo-undertittel"
    >
      VIK FENGSEL
    </h2>
    <p
      className="typo-normal"
    >
      Organisasjonsnummer 974 736 942
    </p>
  </div>
  <hr
    className="box__linje-bred"
  />
  <div
    className="box"
  >
    <div
      className="box__content"
    >
      <ul
        className="list-column-2"
      >
        <li>
          <div
            className="listelement__header"
          >
            <p
              className="typo-element"
            >
              Institusjonstype
            </p>
          </div>
          <div
            className="content "
          >
            FO
          </div>
        </li>
        <li>
          <div
            className="listelement__header"
          >
            <p
              className="typo-element"
            >
              Periode
            </p>
          </div>
          <div
            className="content "
          >
            <div
              className="inst__periode"
            >
              13.06.2017 - 15.09.2017
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
